File: Data Structures & Algorithms/two-integer-sum/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> map = new Dictionary<int, int>();

        for (int i = 0; i < nums.Count(); i++)
        {
            if (map.ContainsKey(target - nums[i]))
            {
                return [map[target - nums[i]], i];
            }

            map.Add(nums[i], i);
        }

        return [0, 0];
    }
}

File: Data Structures & Algorithms/binary-search/submission-2.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;

        int mid = (r - l) / 2 + l;
        while (l <= r)
        {
            if (nums[mid] == target) return mid;
            else if (nums[mid] < target)
            {
                l = mid + 1;
            }
            else r = mid - 1;

            mid = (r - l) / 2 + l;
        }

        return -1;
    }
};
